onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1686,21 +1702,41 @@ Status GemmOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
   bool has_bias = inputs.size() > 2;
   if (has_bias) {
     const auto& bias = inputs[2].node_arg.Name();
-    // We need squeeze the input tensor to 1d if necessary
-    if (shaper[bias].size() > 1) {
-      std::string bias_squeezed = model_builder.GetUniqueName(node_unit.Name() + op + "_bias_squeezed");
-      // We will use squeeze all here
-      ORT_RETURN_IF_ERROR(AddSqueezeOp(model_builder, node_unit.Name(),
-                                       bias, bias_squeezed,
-                                       {} /* axes */));
-      bias_idx = operand_indices.at(bias_squeezed);
-      LOGS_DEFAULT(VERBOSE) << "GemmOpBuilder - Operand [" << bias << "] squeezed from "
-                            << Shape2String(shaper[bias])
-                            << " to "
-                            << Shape2String(shaper[bias_squeezed]);
-    } else {
+    if (!is_quant_gemm) {
+      // We need squeeze the input tensor to 1d if necessary
+      if (shaper[bias].size() > 1) {
+        std::string bias_squeezed = model_builder.GetUniqueName(node_unit.Name() + op + "_bias_squeezed");
+        // We will use squeeze all here
+        ORT_RETURN_IF_ERROR(AddSqueezeOp(model_builder, node_unit.Name(),
+                                         bias, bias_squeezed,
+                                         {} /* axes */));
+        bias_idx = operand_indices.at(bias_squeezed);
+        LOGS_DEFAULT(VERBOSE) << "GemmOpBuilder - Operand [" << bias << "] squeezed from "
+                              << Shape2String(shaper[bias])
+                              << " to "
+                              << Shape2String(shaper[bias_squeezed]);
+      } else {
+        bias_idx = operand_indices.at(bias);
+      }
+    } else {  // is_quant_gemm
+      if (is_quant_gemm) {
```

Comment:
>if (is_quant_gemm) { [](http://example.com/codeflow?start=6&length=20)

looks like this condition is not necessary since we are already in the else block of `!is_quant_gemm`